### PR TITLE
feat: support webpack5 and html-webpack-plugin5

### PR DESCRIPTION
**Pull request by githoniel**
RT

Note: It does not support old version of  `webpack` and `html-webpack-plugin`. But if it's need, I could and support for webpack4.

all test pass.

Note: since `html-webpack-plugin` remove chunk info in `beforeAssetTagGeneration` hook，`doesChunkBelongToHTML`  refactor to use chunkName filter. I had add a test for it.

### CODE REVIEW

**Comment by githoniel** [comment]
@sodatea I think @vue/cli@5.0.0 may need this

**Comment by skyrex-mark** [comment]
any updates on this pr?

**Review by sodatea — CHANGES_REQUESTED** [review]
Thank you so much for the help!

A few things, though:
1. `yarn.lock` contains private registry URLs, please fix
2. It would be better if webpack 4 can still be supported. (It's okay to drop support if that's not easy)

**Comment by githoniel** [comment]
> Thank you so much for the help!
> 
> A few things, though:
> 
> 1. `yarn.lock` contains private registry URLs, please fix
> 2. It would be better if webpack 4 can still be supported. (It's okay to drop support if that's not easy)

ok i will fix both

File: package.json [hunk 1]
```diff
@@ -36,16 +37,17 @@
...
     "memory-fs": "^0.4.1",
     "nyc": "^11.7.1",
     "rimraf": "^2.6.2",
-    "webpack": "^5.4.0"
+    "webpack": "^5.20.0"
   },
   "peerDependencies": {
-    "html-webpack-plugin": ">=4.5.0",
-    "webpack": ">=5.4.0"
+    "html-webpack-plugin": ">=5.0.0 || ^^4.5.1",
```
Comment (sodatea) [inline package.json:50]:
```suggestion
    "html-webpack-plugin": ">=5.0.0 || ^4.5.1",
```

**Review by sodatea — APPROVED** [review]
Looks good to me. Thanks!

**Comment by skyrex-mark** [comment]
Thank you for all the hard work on adding w5 support!

**Comment by skyrex-mark** [comment]
It seems like there is no release for this change yet. Is there a plan to release this to npm soon? @sodatea

**Comment by sodatea** [comment]
It's Lunar New Year holiday here… I'll release the new version when I have the time.

**Comment by sodatea** [comment]
@skyrex-mark Released as v2.0.0

**Pull request merged**